packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -201,7 +201,8 @@ export class AuthUtil {
             conn = await this.auth.reauthenticate(conn)
         }
 
-        return this.secondaryAuth.useNewConnection(conn)
+        await this.secondaryAuth.useNewConnection(conn)
+        return conn
     }
 
     public async connectToEnterpriseSso(startUrl: string, region: string) {
```

Comment:
Might be worth adding an explicit return type since it has changed from `void` to `SsoConnection`